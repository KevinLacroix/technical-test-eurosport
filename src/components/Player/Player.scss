.Player {
  padding: 20px;
  background: gainsboro;
  text-align: center;
}

.Player__header {
  margin-bottom: 20px;
}

.Player__picture {
  position: relative;
  width: 100px;
  height: 100px;
  overflow: hidden;
  border-radius: 50%;
  margin: 0 auto 20px auto;
}

.Player__img {
  position: absolute;
  top: 0;
  left: 0;
}

.Player__name {
  margin: 0 0 10px 0;
  font-size: 30px;
}

.Player__origin {
  display: flex;
  justify-content: center;
}

.Player__imgFlag {
  width: 24px;
  height: auto;
  margin-right: 15px;
}

.Player__country {
  font-size: 16px;
}

.Player__listDetails {
  display: flex;
  padding: 0;
  margin: 0 0 20px 0;
  list-style: none;
  justify-content: space-between;
  align-items: flex-end;
}

.Player__detailLabel {
  display: block;
}

.Player__detailValue {
  display: block;

  &.Player__detailValue--lg {
    font-size: 40px;
    font-weight: 700;
  }
}
