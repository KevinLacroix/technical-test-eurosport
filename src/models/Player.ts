export interface IPlayer {
  id: number;
  firstname: string;
  lastname: string;
  flag: string;
  countryCode: string;
  picture: string;
  rank: number;
  weight: number;
  height: number;
  age: number;
  lastgames: number[];
}

export class Player implements IPlayer {
  id: number;
  firstname: string;
  lastname: string;
  flag: string;
  countryCode: string;
  picture: string;
  rank: number;
  weight: number;
  height: number;
  age: number;
  lastgames: number[];

  constructor(id: number) {
    this.id = id;
    this.firstname = '';
    this.lastname = '';
    this.flag = '';
    this.countryCode = '';
    this.picture = '';
    this.rank = 0;
    this.weight = 0;
    this.height = 0;
    this.age = 0;
    this.lastgames = [];
  }
}
