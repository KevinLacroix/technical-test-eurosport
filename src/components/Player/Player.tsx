import React from 'react';
import classnames from 'classnames';
import { countryCodes } from './../../configurations/countryCodes';
import { formatWeight, formatHeight } from './../../utils';
import './Player.scss';

export interface IPlayerProps {
  firstname: string;
  lastname: string;
  flag: string;
  countryCode: string;
  picture: string;
  rank: number;
  weight: number;
  height: number;
  age: number;
  lastgames: number[];
  className: string;
}

export const Player: React.FC<IPlayerProps> = props => {
  const country = countryCodes[`${props.countryCode.toLowerCase()}`];
  const nbVictory = props.lastgames.filter(game => game === 1).length;
  const nbDefeat = props.lastgames.filter(game => game === 0).length;

  return (
    <article className={classnames('Player', props.className)}>
      <header className="Player__header">
        <div className="Player__picture">
          <img
            src={props.picture}
            alt={`${props.firstname} ${props.lastname}`}
            className="Player__img"
          />
        </div>
        <h3 className="Player__name">
          {props.firstname} {props.lastname}
        </h3>
        <div className="Player__origin">
          <img src={props.flag} alt={country} className="Player__imgFlag" />
          <div className="Player__country">{country}</div>
        </div>
      </header>
      <div className="Player__body">
        <ul className="Player__listDetails">
          <li className="Player__detail">
            <span className="Player__detailLabel">Poids (Kg)</span>
            <span className="Player__detailValue">
              {formatWeight(props.weight)}
            </span>
          </li>
          <li className="Player__detail">
            <span className="Player__detailLabel">Taille</span>
            <span className="Player__detailValue">
              {formatHeight(props.height)}m
            </span>
          </li>
          <li className="Player__detail">
            <span className="Player__detailLabel">Age</span>
            <span className="Player__detailValue">{props.age}</span>
          </li>
          <li className="Player__detail">
            <span className="Player__detailLabel">Classement</span>
            <span className="Player__detailValue Player__detailValue--lg">
              {props.rank}
            </span>
          </li>
        </ul>
        <div>Nombre de victoires: {nbVictory}</div>
        <div>Nombre de d√©faites: {nbDefeat}</div>
      </div>
    </article>
  );
};
