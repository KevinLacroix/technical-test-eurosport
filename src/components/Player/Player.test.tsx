import React from 'react';
import { mount, ReactWrapper, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Player as PlayerModel } from './../../models/Player';
import { Player as PlayerComponent } from './../../components/Player';

configure({ adapter: new Adapter() });

let player: PlayerModel;
let Player: ReactWrapper;

beforeEach(() => {
  player = {
    id: 1,
    firstname: 'Stan',
    lastname: 'Wawrinka',
    flag: 'https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png',
    countryCode: 'SUI',
    picture:
      'https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg',
    rank: 21,
    weight: 81000,
    height: 183,
    age: 33,
    lastgames: [1, 1, 1, 0, 1]
  };
});

afterEach(() => {
  Player.unmount();
});

describe('Player Component', () => {
  test('render', () => {
    Player = mount(
      <PlayerComponent
        firstname={player.firstname}
        lastname={player.lastname}
        flag={player.flag}
        countryCode={player.countryCode}
        picture={player.picture}
        rank={player.rank}
        weight={player.weight}
        height={player.height}
        age={player.age}
        lastgames={player.lastgames}
        className="player-test"
      />
    );

    expect(Player).toMatchSnapshot();
  });
});
