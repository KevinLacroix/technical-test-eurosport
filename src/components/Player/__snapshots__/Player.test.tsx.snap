// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player Component render 1`] = `ReactWrapper {}`;
